app/models/archived/petition.rb

```diff
@@ -216,6 +216,10 @@ def removed?(id)
         removed.exists?(id)
       end
 
+      def can_anonymize?
+        where(anonymized_at: nil).where(do_not_anonymize: [nil, false]).any?
```

Comment:
We're going to need to check this query in production for performance - hundreds of thousands of rows in the archived_petitions table. It's probably okay since there's a `LIMIT 1` so PostgreSQL should just check for the existence of one row and stop but may need an index just to give it a helping hand.